### PR TITLE
Added a Color Wheel for Harmony Color Modes

### PR DESCRIPTION
**Pull request by riggaroo**
Added a new kind of color wheel for getting different `ColorHarmonyModes`. 

# Usage

```
HarmonyColorPicker(
            harmonyMode = ColorHarmonyMode.COMPLEMENTARY,
            modifier = Modifier.size(400.dp),
            onColorChanged = { hsvColor ->
                currentColor.value = hsvColor.toColor()
           })
```

# Screenshots

Android:
Screenshot_1642751332
Screenshot_1642751327
Screenshot_1642751323

Desktop:

Screenshot 2022-01-21 at 09 46 13

### CODE REVIEW

File: color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorMagnifiers.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        animationSpec = spring(dampingRatio = Spring.DampingRatioLowBouncy)
+    )
+
+    val diameterDp = with(LocalDensity.current) {
+        Dp(this.density * diameterPx)
+    }
+
+    val animatedDiameter = animateDpAsState(
+        targetValue = if (currentlyDragging) {
+            diameterDp * 0.022f
```
Comment (sbasheleyshvili-godaddy) [inline color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorMagnifiers.kt:38]: [nit] A named constant would be nice here

File: color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorMagnifiers.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        } else {
+            diameterDp * 0.020f
```
Comment (sbasheleyshvili-godaddy) [inline color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorMagnifiers.kt:40]: [nit] And here

File: color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorMagnifiers.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }
+    )
+
+    hsvColor.getColors(harmonyMode).forEach { color ->
+        val positionForColor by animateOffsetAsState(
+            targetValue = positionForColor(color, size),
+            animationSpec = spring(dampingRatio = Spring.DampingRatioLowBouncy)
+        )
+        Magnifier(position = positionForColor, color = color, diameter = diameterDp * 0.015f)
```
Comment (sbasheleyshvili-godaddy) [inline color-picker/src/commonMain/kotlin/com/godaddy/android/colorpicker/harmony/HarmonyColorMagnifiers.kt:49]: [nit] And here as well

**Review by sbasheleyshvili-godaddy — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/godaddy/android/colorpicker/ColorPreviewInfo.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun ColorPreviewInfo(currentColor: Color) {
+    Column(modifier = Modifier.fillMaxWidth()) {
+        Text(
+            modifier = Modifier.padding(16.dp),
+            text = "a: ${currentColor.alpha} \n" +
```
[the remark this answers is not in the document]
Comment (riggaroo) [reply app/src/main/java/com/godaddy/android/colorpicker/ColorPreviewInfo.kt:23]: Yeah this is just the sample app, not the library.

File: app/src/main/java/com/godaddy/android/colorpicker/HarmonyColorPickerScreen.kt [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                    Text(it.name, modifier = Modifier.clickable {
+                        harmonyMode.value = it
+                        expanded.value = false
+                    })
+                }
+            }
+        }
+        HarmonyColorPicker(
+            harmonyMode = harmonyMode.value,
+            modifier = Modifier.size(400.dp),
```
[the remark this answers is not in the document]
Comment (riggaroo) [reply app/src/main/java/com/godaddy/android/colorpicker/HarmonyColorPickerScreen.kt:59]: This is just the sample app showing how you can set the size.

File: app/src/main/java/com/godaddy/android/colorpicker/SampleColorPickerActivity.kt [hunk 6]
```diff
@@ -63,53 +46,24 @@ class SampleColorPickerActivity : ComponentActivity() {
...
-        Text(
-            modifier = Modifier.padding(16.dp),
-            text = "a: ${currentColor.alpha} \n" +
-                    "r: ${currentColor.red} \n" +
-                    "g: ${currentColor.green} \n" +
-                    "b: ${currentColor.blue}"
+        TopAppBar(title = {
+            Text(stringResource(R.string.compose_color_picker_sample))
+        })
+        Text("Classic Picker",
```
[the remark this answers is not in the document]
Comment (riggaroo) [reply app/src/main/java/com/godaddy/android/colorpicker/SampleColorPickerActivity.kt:54]: This is just the demo app. The library doesn't have any strings.

File: app/src/main/java/com/godaddy/android/colorpicker/SampleColorPickerActivity.kt [hunk 7]
```diff
@@ -63,53 +46,24 @@ class SampleColorPickerActivity : ComponentActivity() {
...
+            modifier = Modifier.defaultMinSize(minHeight = 48.dp)
+                .fillMaxWidth()
+                .clickable {
+                    navController.navigate("classic")
```
[the remark this answers is not in the document]
Comment (riggaroo) [reply app/src/main/java/com/godaddy/android/colorpicker/SampleColorPickerActivity.kt]: Done

File: app/src/main/java/com/godaddy/android/colorpicker/ClassicColorPickerScreen.kt [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+@Composable
+fun ClassicColorPickerScreen(navController: NavController) {
+    Column {
+        TopAppBar(title = {
+            Text(stringResource(R.string.classic_color_picker_sample))
+        },
+            navigationIcon = {
+                BackButton { navController.navigateUp() }
+            })
```
Comment (sbasheleyshvili-godaddy) [inline app/src/main/java/com/godaddy/android/colorpicker/ClassicColorPickerScreen.kt:27]: [nit] Something's off with the formatting here

File: README.md [hunk 9]
```diff
@@ -33,13 +36,29 @@ Column {
 }
 ```
 
+Or Add the `HarmonyColorPicker` to your Compose hierarchy for a HSV circle color wheel implementation:
```
Comment (sbasheleyshvili-godaddy) [inline README.md:39]: [nit] 'HSV circle color wheel' seems a bit redundant. I'd remove 'circle'.

File: README.md [hunk 10]
```diff
@@ -33,13 +36,29 @@ Column {
...
+ HarmonyColorPicker(
+    harmonyMode = harmonyMode.value,
+    modifier = Modifier.size(400.dp),
+    onColorChanged = { hsvColor ->
+        currentColor.value = hsvColor.toColor()
+        extraColors.value = hsvColor.getColors(colorHarmonyMode = harmonyMode.value) 
+})
+```
+
+The `HarmonyColorPicker` allows for you to set a certain `ColorHarmonyMode` to enable the different options to be
```
Comment (sbasheleyshvili-godaddy) [inline README.md:51]: [nit] I'd rephrase this. It's not just 'options displayed', it's a whole new feature that deserves a proper description.

File: README.md [hunk 11]
```diff
@@ -65,3 +84,46 @@ ClassicColorPicker(
...
+    modifier = Modifier.height(200.dp),
+    onColorChanged = { color: HsvColor ->
+        // Do something with the color
+    }
+)
+```
+
+### Harmony Mode
+
+To Change the harmony mode of the picker, pass in a different mode into the function:
```
Comment (sbasheleyshvili-godaddy) [inline README.md:108]: [nit] Odd capitalization of 'Change'.

**Review by sbasheleyshvili-godaddy — APPROVED** [review]
